From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Mon, 17 Dec 2018 20:38:41 +0100
Subject: media: sun6i-csi: Add support for missing 16bit color formats

CSI engine will not try to reorder the color parts, so this is safe
for matching mbus formats.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/media/platform/sunxi/sun6i-csi/sun6i_csi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/sunxi/sun6i-csi/sun6i_csi.c b/drivers/media/platform/sunxi/sun6i-csi/sun6i_csi.c
index 111111111111..222222222222 100644
--- a/drivers/media/platform/sunxi/sun6i-csi/sun6i_csi.c
+++ b/drivers/media/platform/sunxi/sun6i-csi/sun6i_csi.c
@@ -316,6 +316,9 @@ static enum csi_input_seq get_csi_input_seq(struct sun6i_csi_device *csi_dev,
 		break;
 
 	case V4L2_PIX_FMT_YUYV:
+	case V4L2_PIX_FMT_YVYU:
+	case V4L2_PIX_FMT_UYVY:
+	case V4L2_PIX_FMT_VYUY:
 		return CSI_INPUT_SEQ_YUYV;
 
 	default:
-- 
Armbian

